libmount/src/context_mount.c

```diff
@@ -589,13 +593,20 @@ static int exec_helper(struct libmnt_context *cxt)
 	if (rc)
 		return -EINVAL;
 
+	if (cxt->namespaces[MNT_NS_TGT].fd != -1
```

Comment:
What about to add mnt_context_has_namespace() rather than use cxt->namespaces[MNT_NS_TGT].fd in code?